### PR TITLE
Checkbox common form props

### PR DESCRIPTION
**Pull request by JoMarton**
After updating the common FormInputProps interface, this PR:
1. Removes redundant props from the component props list
2.Adds tests and implementation for props that weren't addressed previously

### CODE REVIEW

File: test/components/checkbox.spec.tsx [hunk 1]
```diff
@@ -193,6 +192,56 @@ describe('<Checkbox/>', () => {
...
+    it('Accepts "autofocus" prop', async () => {
+        if (document.hasFocus()) {
+
+            const {driver: checkbox, waitForDom} = clientRenderer.render(
+                <CheckBox  autoFocus/>
+            ).withDriver(CheckBoxTestDriver);
+
+            await waitForDom(() => {
+                expect(document.activeElement).to.equal(checkbox.nativeInput);
+                expect(checkbox.elementHasStylableState('focus')).to.equal(true);
```
Comment (alisey) [inline test/components/checkbox.spec.tsx]: We're missing a similar test for `:disabled` state, although we have a test which ensures that the native input receives the attribute.

Comment (JoMarton) [reply test/components/checkbox.spec.tsx]: added a test

File: src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -65,34 +57,37 @@ export class CheckBox extends React.Component<CheckBoxProps, CheckBoxState> {
                 onClick={this.handleChange}
                 style-state={styleState}
                 role="checkbox"
-                aria-checked={indeterminate ? 'mixed' : value}
+                aria-checked={this.props.indeterminate ? 'mixed' : this.props.value}
             >
                 <input
```
Comment (Zombiefruit) [inline src/components/checkbox/checkbox.tsx]: Seems to be missing the required prop? There should also be a test for this

Comment (JoMarton) [reply src/components/checkbox/checkbox.tsx]: This was actually on purpose... we need to have some product discussions regarding `required` behavior and validation in general

**Pull request merged**